src/app/events/tracking/HandleParticipantDuringEvent.ts

```diff
@@ -47,7 +68,43 @@ const HandleParticipantDuringEvent = async (oldState: VoiceState, newState: Voic
 	
 	if (hasUserChangedChannels(oldState, newState)) {
 		if (await isChannelActivePOAPEvent(oldState.channelId, oldState.guild.id)) {
+			const db: Db = await MongoDbUtils.connect(constants.DB_NAME_DEGEN);
+			const addtionalPoapParticipantsCol: Collection<POAPParticipant> = db.collection(constants.DB_COLLECTION_ADDITIONAL_POAP_PARTICIPANTS_LIST);
+			const leavingUser = retrieveParticipantFromAdditionalList({ id: oldState.id, tag: oldState.member?.user.tag }, oldState.channelId, oldState.guild.id);
+			const leavingUserQuery: POAPParticipant | null = await addtionalPoapParticipantsCol.findOne({
+				discordUserId: oldState.id,
+			});
+			const participant = await retrieveActiveParticipant({ id: oldState.id, tag: oldState.member?.user.tag }, oldState.channelId, `${oldState.member?.user.tag}`);
+			if (!leavingUserQuery) {
+				const minutesAttended = dayjs().diff(participant?.startTime, 'm', true);
+				const resultInsert: InsertOneWriteOpResult<POAPParticipant> | void = await addtionalPoapParticipantsCol.insertOne({
+					discordUserId: oldState.id,
+					discordUserTag: oldState.member?.user.tag,
+					voiceChannelId: oldState.channelId,
+					minutesAttended:  minutesAttended,
+					timeLeft: dayjs().toISOString(),
```

Comment:
is timeLeft the current date in ISO?

Reply:
yes

Reply:
it saves the time a user leaves the voice channel